File: client/src/app/core/dto/contact/contact.dto.ts
import { Guid } from 'guid-typescript';
import { StageStatusEnum } from '../../enums/contact/stage-status.enum';
import { ContactTypeEnum } from '../../enums/contact/contact-type.enum';

export class ContactDto {
  Id: Guid = Guid.createEmpty();
  Name: string = '';
  Notes?: string;
  Email?: string;
  Phone1?: string;
  Phone2?: string;
  Phone3?: string;
  ContactType: ContactTypeEnum = ContactTypeEnum.None;
  StageStatus?: StageStatusEnum;
}

File: api/Core/Dto/Contact/ContactDto.cs
using ProductivityHarborApi.Core.Enums.Contact;

namespace ProductivityHarborApi.Core.Dto.Contact
{
    public class ContactDto
    {
        public Guid Id { get; set; }
        public string Name { get; set; } = string.Empty;
        public string? Notes { get; set; }
        public string? Email { get; set; }
        public string? Phone1 { get; set; }
        public string? Phone2 { get; set; }
        public string? Phone3 { get; set; }
        public ContactTypeEnum ContactType { get; set; }
        public StageStatusEnum? StageStatus { get; set; }
        public virtual DateTimeOffset CreatedAt { get; set; } = DateTimeOffset.Now;
        public virtual Guid? CreatedByUserId { get; set; }
        public virtual DateTimeOffset? UpdatedAt { get; set; }
        public virtual Guid? UpdatedByUserId { get; set; }
    }
}
